nestkernel/dynamicloader.cpp

```diff
@@ -270,7 +270,11 @@ DynamicLoaderModule::init( SLIInterpreter* i )
     LOG( M_ERROR, "DynamicLoaderModule::init", "Could not initialize libltdl. No dynamic modules will be available." );
   }
 
-  if ( lt_dladdsearchdir( NEST_INSTALL_PREFIX "/" NEST_INSTALL_LIBDIR ) )
+  // Prevent compile-time construction of path to avoid incorrect Conda replacement. See #2237.
+  std::string moduledir = NEST_INSTALL_PREFIX;
```

Comment:
Instead of the changes suggested by @jougs below, I'd just do this here:
```suggestion
  std::string moduledir = std::string( NEST_INSTALL_PREFIX ).c_str();
```
The logic is explained here: https://github.com/conda/conda-build/issues/1674#issuecomment-280378336